src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -0,0 +1,174 @@
+        args.txParams.data,
+        args.txParams.operation,
+        args.txParams.safeTxGas,
+        args.txParams.baseGas,
+        args.txParams.gasPrice,
+        args.txParams.gasToken,
+        args.txParams.refundReceiver,
+        sigs,
+      ])
+
+      return {
+        operation: OperationType.Call,
+        to: safeAddress,
+        value: '0',
+        data,
+      }
+    })
+    .filter(Boolean) as MetaTransactionData[]
+}
+
+const getTxsWithDetails = (txs: Transaction[], chainId: string) => {
+  return Promise.all(
+    txs.map(async (tx) => {
+      return await getTransactionDetails(chainId, tx.transaction.id)
+    }),
+  )
+}
+
+const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
+  const [multiSendContract, setMultiSendContract] = useState<Multi_send_call_only>()
+  const [multiSendTxData, setMultiSendTxData] = useState<string>()
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chain = useCurrentChain()
+  const { safe } = useSafeInfo()
+  const provider = useWeb3()
+
+  const [txsWithDetails, error, loading] = useAsync<TransactionDetails[]>(() => {
+    if (!chain?.chainId) return
+
+    return getTxsWithDetails(data.txs, chain.chainId)
+  }, [data.txs, chain?.chainId])
+
+  useEffect(() => {
+    if (!chain) return
+
+    setMultiSendContract(getMultiSendCallOnlyContractInstance(chain.chainId))
+  }, [chain])
+
+  useEffect(() => {
+    if (!txsWithDetails || !chain) return
+
+    const multiSendTxs = getMultiSendTxs(txsWithDetails, chain, safe.address.value, safe.version)
+    const data = encodeMultiSendData(multiSendTxs)
+
+    setMultiSendTxData(data)
+  }, [chain, safe.address.value, safe.version, txsWithDetails])
+
+  const sendTx = async () => {
+    setSubmitError(undefined)
+    if (!provider || !multiSendTxData || !chain || !multiSendContract || !txsWithDetails) return
+
+    try {
+      const multiSendTransaction = await multiSendContract.connect(provider.getSigner()).multiSend(multiSendTxData)
+
+      txsWithDetails.forEach((tx) => {
+        createExistingTx(chain.chainId, safe.address.value, tx.txId, tx).then((safeTx) =>
+          txDispatch(TxEvent.MINING, { txId: tx.txId, txHash: multiSendTransaction.hash, tx: safeTx }),
+        )
+      })
+
+      onSubmit(null)
+    } catch (err) {
+      logError(Errors._804, (err as Error).message)
+      setSubmitError(err as Error)
+    }
+  }
+
+  return (
+    <div>
+      <DialogContent>
+        <Typography variant="body2" mb={2}>
+          This transaction batches a total of {data.txs.length} transactions from your queue into a single Ethereum
+          transaction. Please check every included transaction carefully, especially if you have rejection transactions,
+          and make sure you want to execute all of them. Included transactions are highlighted in green when you hover
+          over the execute button.
+        </Typography>
```

Comment:
Please add some newlines between the logical blocks in the JSX, it's a bit hard to read otherwise.